extensions/amp-video-docking/0.1/amp-video-docking.js

```diff
@@ -594,8 +596,9 @@ export class VideoDocking {
     }
 
     const {element} = video;
-    const {intersectionRect} = element.getIntersectionChangeEntry();
-    if (!isSizedRect(intersectionRect)) {
+    const clientRect = element.getBoundingClientRect();
+    const intersectionRect = rectIntersection(clientRect, this.viewportRect_);
+    if (!intersectionRect || !isSizedRect(intersectionRect)) {
```

Comment:
We don't have a better way to say "is element within viewport"?

Reply:
No better "sync" way. I tried to keep the structure as much as possible intact. There's a v2 version of this API in works and we'll need to make sure we just use InOb for most of the needs there.